services/stats/memstats/stats.go

```diff
@@ -0,0 +1,181 @@
+package memstats
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/services/stats"
+)
+
+var _ stats.Stats = (*Store)(nil)
+
+var _ stats.Measurement = (*Measurement)(nil)
+
+type Store struct {
+	once  sync.Once
+	mu    sync.Mutex
+	byKey map[string]*Measurement
+
+	Now func() time.Time
+}
+
+type Measurement struct {
+	mu        sync.Mutex
+	startTime time.Time
+	now       func() time.Time
+
+	Tags stats.Tags
+	Name string
+	Type string
+
+	LastValue    float64
+	Values       []float64
+	LastDuration time.Duration
+	Durations    []time.Duration
+}
+
+// Count implements stats.Measurement
+func (m *Measurement) Count(n int) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if m.Type != stats.CountType {
+		panic("operation Count not supported for measurement type:" + m.Type)
+	}
+
+	m.LastValue += float64(n)
+	m.Values = append(m.Values, m.LastValue)
+}
+
+// Increment implements stats.Measurement
+func (m *Measurement) Increment() {
+	if m.Type != stats.CountType {
+		panic("operation Increment not supported for measurement type:" + m.Type)
+	}
+
```

Comment:
This check is redundant.

```suggestion
```

Reply:
Why do you say this check is redundant?

Reply:
I see; I did it this way cause the panic would be a bit misleading otherwise.

```go
m.Increment()
```

Will panic with `operation Count not supported for measurement type:" + m.Type`